accname/name/comp_hidden_not_referenced.html

```diff
@@ -11,6 +11,31 @@
 </head>
 <body>
 
+<!-- Buttons hidden by assorted techniques -->
+<button class="ex" data-testname="hidden button" data-expectedlabel="" id="Apple" hidden>Apple</button>
+<button class="ex" data-testname="display:none button" data-expectedlabel="" id="Blueberry" style="display:none;">Blueberry</button>
+<button class="ex" data-testname="visibility:hidden button" data-expectedlabel="" id="Cantaloupe" style="visibility:hidden;">Cantaloupe</button>
+<button class="ex" data-testname="visibility:collapse button" data-expectedlabel="" id="Durian" style="visibility:collapse;">Durian</button>
+<button class="ex" data-testname="content-visibility:hidden button" data-expectedlabel="" id="Eggplant" style="content-visibility:hidden;">Eggplant</button>
+<button class="ex" data-testname="aria-hidden=true button" data-expectedlabel="" id="Fig" aria-hidden="true">Fig</button>
+
+<!-- Hidden buttons which are labeled by other hidden buttons -->
+<button class="ex" data-testname="hidden button, labelledby hidden button" data-expectedlabel="" id="Grape" hidden aria-labelledby="Apple">Grape</button>
+<button class="ex" data-testname="display:none button, labelledby display:none button" data-expectedlabel="" id="Honeydew" style="display:none;" aria-labelledby="Blueberry">Honeydew</button>
+<button class="ex" data-testname="visibility:hidden button, labelledby visibility:hidden button" data-expectedlabel="" id="Itapalm" style="visibility:hidden;" aria-labelledby="Cantaloupe">Itapalm</button>
+<button class="ex" data-testname="visibility:collapse button, labelledby visibility:collapse button" data-expectedlabel="" id="Jujube" style="visibility:collapse;" aria-labelledby="Durian">Jujube</button>
+<button class="ex" data-testname="content-visibility:hidden button, labelledby content-visibility:hidden button" data-expectedlabel="" id="Kiwi" style="content-visibility:hidden;" aria-labelledby="Eggplant">Kiwi</button>
+<button class="ex" data-testname="aria-hidden=true button, labelledby aria-hidden=true button" data-expectedlabel="" id="Lychee" aria-hidden="true" aria-labelledby="Fig">Lychee</button>
+
+<!-- Non-hidden buttons which are labelled by hidden buttons -->
```

Comment:
Non-hidden elements labeled by hidden elements is plausible, but not so much with buttons referencing other hidden buttons... Here are some potential examples:

button "Do it!" with hidden referenced label and another hidden but non-referenced node
```html
<button aria-labelledby="button_label">
  <!-- no rendered text, but maybe a background image or some other anti-pattern -->
  <span hidden id="button_label">Do it!</span>
  <span hidden id="not_referenced"> but this should not be in the label </span>
</button>
```

link "More Info about Widgets" with other referenced and non-referenced hidden content
```html
<a href="#" aria-labelledby="link_text link_hidden_text">
  <span id="link_text">
    More Info
    <span hidden> hidden </span>
  </span>
  <span hidden> hidden </span>
  <span hidden id="link_hidden_text"> about Widgets </span>
</button>
```

@aleventhal or @jaragunde probably wrote a bunch more "hidden not referenced" test cases for Chromium and may be able to share a direct link. 

I also recall discussions about inverting `aria-hidden=true` with a descendant `aria-hidden=false` in a visibly rendered node, and similar for inverting CSS `visibility: hidden;` with `visibility: visible;` in a descendant node, though I can't recall where WebKit, Gecko, and Chromium stand on those edge cases.

Reply:
h1 label "not hidden not hidden" IIRC
```html
<h1>not hidden
  <div style="visibility: hidden;">
    hidden not referenced
    <span style="visibility: visible;">not hidden</div>
  </div>
</h1> 
```

ditto for aria-hidden, IIRC
```html
<h1>not hidden
  <div aria-hidden="true">
    hidden not referenced
    <span aria-hidden="false">not hidden</div>
  </div>
</h1> 
```

whatever the result: we've seen web devs try things like this, so would be good to get interop.

Reply:
Ooh, and this one. expected label: "not hidden"

```html
<h1>not hidden
  <div hidden>
    hidden not referenced (not rendered)
    <span aria-hidden="false">
        hidden not referenced (aria-hidden=false but still hidden b/c not rendered)
        <!-- there will be some interop differences here -->
    </div>

  </div>
</h1> 
```